backend/tests/clubs/test_ticketing.py

```diff
@@ -525,7 +554,48 @@ def test_get_cart(self):
         self.assertEqual(len(data["tickets"]), 5, data)
         for t1, t2 in zip(data["tickets"], tickets_to_add):
             self.assertEqual(t1["id"], str(t2.id))
-        self.assertEqual(data["sold_out"], 0, data)
+        num_sold_out = sum(item["count"] for item in data["sold_out"])
+        self.assertEqual(num_sold_out, 0, data)
+
+    def test_calculate_cart_total(self):
+        # Add a few tickets to cart
+        cart, _ = Cart.objects.get_or_create(owner=self.user1)
+        tickets_to_add = self.tickets1[:5]
+        for ticket in tickets_to_add:
+            cart.tickets.add(ticket)
+        cart.save()
+
+        expected_total = sum(t.price for t in tickets_to_add)
+
+        from clubs.views import TicketViewSet
+
+        actual_total = TicketViewSet()._calculate_cart_total(cart)
+        self.assertEqual(actual_total, expected_total)
+
+    def test_calculate_cart_total_with_group_discount(self):
+        # Create tickets with group discount
+        tickets = [
+            Ticket.objects.create(
+                type="group",
+                event=self.event1,
+                price=10.0,
+                group_size=2,
+                group_discount=0.2,
+            )
+            for _ in range(10)
+        ]
+
+        # Add enough to activate group discount
```

Comment:
Could you also assert the correct cart total before hitting the threshold for group discount?